### PR TITLE
Follow up grant for Sol2Ink

### PR DESCRIPTION
**Pull request by coreggon11**
#### Project Abstract

This is a follow up grant for already delivered [grant](https://github.com/w3f/Grants-Program/pull/960)

Sol2Ink is a development tool that takes code of already existing Solidity file and transpiles it into Rust and ink! code
The generated code needs a double-check by a real developer, to fix some flaws but the hard work of rewriting Solidity to ink! is done.
This eases the process of transition of already existing projects/teams from Ethereum (or other EVM chains) into Dotsama and ink! smart contracts ecosystem
The transpiler aims to use the standards and PSPs of Dotsama ecosystem
A beta version of Sol2Ink is already working for non-complex smart contracts and we aim to make a fully usable app which can transpile any Solidity source code into a working ink! smart contract. 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [X] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [X] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [X] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Some quick initial feedback: Could you change the level to a level 3 grant? Could you sign our latest terms and conditions? And deliverables 0a-0d of the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) are usually mandatory for all milestones, and deliverable 0e at least for the last one.

**Review by Micassa93 — DISMISSED** [review]

**Comment by coreggon11** [comment]
Hey @Noc2 

We updated the grant proposal according to your comment, you can check it again :smile: 

@Micassa93 sorry, I missed the commit by few minutes after your approval, could you re-approve if you are ok with the changes, please? Thank you! :smile:

**Comment by TtomaS7** [comment]
Hello, W3F team! 
Do you have any updates?

File: applications/sol2ink-follow-up.md
Comment (takahser) [inline applications/sol2ink-follow-up.md:231]:
```suggestion
| 2. | Split contract into traits | Currently all logic and storage is in one contract file. This upgrade will generate traits for each contract that the contract is using, splitting logic and storage into separate traits. This will also nicely handle inheritance of Solidity smart contracts |
```

File: applications/sol2ink-follow-up.md
Comment (takahser) [inline applications/sol2ink-follow-up.md:232]: > code which it currently does not support or parses incorrectly

can you specify what kind of code that would be and why it's not supported or parsed incorrectly?

**Review by takahser — CHANGES_REQUESTED** [review]
@TtomaS7 in addition to my inline comments, I'd like to raise the following points:

- As [David already indicated](https://github.com/w3f/Grants-Program/pull/1164#pullrequestreview-1097540537), deliverables 0a-0d of the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) are usually mandatory for all milestones, and deliverable 0e at least for the last one. Could you update the milestones to include the missing deliverables?
- The total costs are $60k while (according to your milestones) the sum should be $96k instead: $36k + $60k = $96k.

**Comment by TtomaS7** [comment]
Hey @takahser 
thank you for reaching out!

We are decided to get the funding for milestone 2 only, as we mentioned in the application. 
Milestone 1 is already done.

So we are requested for 60k$.

File: applications/sol2ink-follow-up.md
Comment (coreggon11) [reply applications/sol2ink-follow-up.md:232]: we found some bugs while using the first version of sol2ink (for example chaining of selectors or updating fields of a struct inside a mapping etc.)

**Comment by TtomaS7** [comment]
@Noc2 hey!
do you have any updates?

**Review by Noc2 — APPROVED** [review]
Sorry for the late reply here. I’m happy to go ahead with it and mark it as ready for review. I will also share it with the rest of the team.

File: applications/sol2ink-follow-up.md
Comment (takahser) [reply applications/sol2ink-follow-up.md:232]: ok, thanks for clearifying

File: applications/sol2ink-follow-up.md
Comment (takahser) [inline applications/sol2ink-follow-up.md:233]: Does this concern the input? If yes, what's the difference between importing from a solidity contract directly vs. importing from a library? Isn't the input in both cases just a solidity file?

**Review by takahser — CHANGES_REQUESTED** [review]
Sorry for the delay here.
I have 2 more questions. Feel free to also update the deliverables to include the information I asked about.

**Review by SBalaguer — DISMISSED** [review]

File: applications/sol2ink-follow-up.md
Comment (coreggon11) [reply applications/sol2ink-follow-up.md:230]: the guide is something like a enhanced documentation, the inline documentation just describes the functions of the program, the guides go deeper into how it works, and also there is a tutorial for usage

File: applications/sol2ink-follow-up.md
Comment (coreggon11) [reply applications/sol2ink-follow-up.md:233]: its just that currently sol2ink looks for contract/ interface definition, it skips library for now because we want to deal with it in a different way than with contracts, meaning if you change the library to a contract, it will be parsed, butnsince a library is not a contract we will have to do it in a different way

**Review by laboon — APPROVED** [review]

File: applications/sol2ink-follow-up.md
Comment (takahser) [reply applications/sol2ink-follow-up.md:230]: Thanks, could you add this information to the delivery?

File: applications/sol2ink-follow-up.md
Comment (takahser) [reply applications/sol2ink-follow-up.md:233]: Thanks, could you integrate this information into the deliverable spec?

**Comment by coreggon11** [comment]
hey @takahser , I have integrated answers to all of your questions into deliverables :smile:

**Review by takahser — DISMISSED** [review]
@coreggon11 thanks for the update! Happy to go ahead with it.
@laboon @SBalaguer @Noc2 FYI - your previous approvals have been dismissed due to the update, feel free to re-approve:)

**Review by Noc2 — DISMISSED** [review]

File: applications/sol2ink-follow-up.md
Comment (semuelle) [reply applications/sol2ink-follow-up.md:232]: Could you update the deliverable spec to include these (and ideally more) details? Information in commit comments are hard to find and in its current form the deliverable would be hard to verify.

**Review by semuelle — CHANGES_REQUESTED** [review]
Happy to approve as well except for the specification of D3. 

Two minor issues: 
- Next time, it would be great if you could formulate the `Project Details` in such a way that it addresses the current application, not the project as a whole. As it is now, it is mostly unrelated to this milestone.
- `ink!` is written with a lowercase `i`.

**Review by SBalaguer — DISMISSED** [review]

File: applications/sol2ink-follow-up.md
Comment (coreggon11) [reply applications/sol2ink-follow-up.md:231]: Hi @semuelle, sorry for the delay, I have described the issues which will be the goals of this milestone 😄

**Review by Noc2 — APPROVED** [review]
Thanks for the update. I’m still happy to go ahead with it. 

Just to double check: you rebranded to 727.ventures? Or are these now two different entities?

**Review by semuelle — APPROVED** [review]
Thanks for the update. Happy to proceed.

**Review by alxs — APPROVED** [review]

**Comment by TtomaS7** [comment]
> Just to double check: you rebranded to 727.ventures? Or are these now two different entities?

Supercolony as you know it will cease to exist in a few weeks. 
The sleeping partner that Markian started Supercolony with no longer shares the vision and mission of our company. Mainly because of a different values' mindset. The best option is to separate at this point. 
and we would like to continue infrastructure development and enhancing Polkadot with new identity now - 727.ventures.

**Comment by Noc2** [comment]
@TtomaS7 Thanks for the update

**Review by SBalaguer — APPROVED** [review]

**Review by laboon — APPROVED** [review]

**Comment by keeganquigley** [comment]
Hi @coreggon11 How is M1 coming along?

**Comment by coreggon11** [comment]
Hi @keeganquigley, I suppose you mean M2 (as M1 was delivered already 😄 )
We had some blockers regarding this, but we should have it ready within 2 weeks.

**Comment by keeganquigley** [comment]
thanks for the update @coreggon11 apologies yes I meant M2!

**Pull request merged**
